src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateTransformations = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    let modifiedState = { ...currentState };
+
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(modifiedState, action.extraData);
+        break;
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
```

Comment:
Ensure that `action.keysToRemove` is an array before iterating over it. You can add a check like `if (Array.isArray(action.keysToRemove))` to prevent potential runtime errors if `keysToRemove` is undefined or not an array.